dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/resources_plugin/github.py

```
@@ -0,0 +1,159 @@
+        return url
+
+    # [start read_file_method]
+    def read_file(self, suf: str):
+        """Get the content of the file.
+
+        The address of the file is the prefix of the resource plugin plus the parameter suf.
+        """
+        path = self.url_join(self.prefix, suf)
+        return self.req(path)
+
+    # [end read_file_method]
+
+    def req(self, path: str):
+        """Send HTTP request, parse response data, and get file content."""
+        headers = {
+            "user-agent": "Mozilla/5.0 (X11; Linux x86_64) "
+            "AppleWebKit/537.36 (KHTML, like Gecko) Chrome/104.0.0.0 Safari/537.36",
+            "Content-Type": "application/json; charset=utf-8",
+        }
+        if self.access_token is not None:
+            headers.setdefault("Authorization", "Bearer %s" % self.access_token)
+        if self.username is not None and self.password is not None:
+            base64string = base64.b64encode(
+                bytes("%s:%s" % (self.username, self.password), encoding="utf-8")
+            )
+            headers.setdefault(
+                "Authorization", "Basic %s" % base64string.decode("utf-8")
+            )
+
+        file_info = self.get_file_info(path)
+        url = self.get_req_url(file_info)
+        params = {"ref": file_info["branch"]}
+        response = requests.get(
+            headers=headers,
+            url=url,
+            params=params,
+        )
+        if response.status_code == requests.codes.ok:
+            json_response = response.json()
+            content = base64.b64decode(json_response["content"])
+            return content.decode("utf-8")
+        else:
+            if response.status_code == requests.codes.not_found:
+                raise PyResPluginException("{} is not found.".format(path))
+            if response.status_code == requests.codes.unauthorized:
+                raise PyResPluginException("unauthorized.")
+            raise PyResPluginException("Unknown exception.")
```

we should raise the original error message from response